core/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -244,5 +244,13 @@ private void resolvePipelineAggregatorOrder(Map<String, AggregatorFactory> aggFa
                 orderedPipelineAggregators.add(factory);
             }
         }
+
+        AggregatorFactory[] getAggregatorFactories() {
+            return this.factories.toArray(new AggregatorFactory[this.factories.size()]);
+        }
+
+        List<PipelineAggregatorFactory> getPipelineAggregatorFactories() {
```

Comment:
Why do we need to convert to an array? In general I think we should try to use either lists or arrays everywhere instead of doing back-and-forth conversions? (not for the cost of the conversions, more for consistency) Maybe this should be a different PR though, it looks like an existing issue to me. It was just weird when looking at the diff